### PR TITLE
[TRAFODION-3008] support a new TMUDF to generate a series of Number

### PR DESCRIPTION
**Pull request by andyyangcn**

### CODE REVIEW

**Comment by sureshsubbiah** [comment]
+1. This is a nice UDF to have. I wonder if we can somehow get a parallel plan for this UDF, which may cause repeated values. This is a minor and likely unnecessary concern.

File: core/sql/sqludr/SqlUdrPredefSeries.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+extern "C" UDR * SERIES()
+{
+  return new series();
+}
+
+/*  Step 3: Write the actual UDF code*/
+void series::describeParamsAndColumns(
+                           UDRInvocationInfo &info)
+{
+  // We always expect one input parameter
```
Comment (DaveBirdsall) [inline core/sql/sqludr/SqlUdrPredefSeries.cpp]: The comment is incorrect; from the code it looks like we expect either two or three input parameters.

File: core/sql/sqludr/SqlUdrPredefSeries.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    throw UDRException(38001,
+                       "Expecting two or three input parameters");
+
+  if (info.par().getType(0).getSQLTypeClass() != info.par().getType(1).getSQLTypeClass())
+    throw UDRException(38001,
+                       "The data type of the first two parameters should be same");
+
+  if (info.par().getType(0).getSQLTypeClass() != TypeInfo::NUMERIC_TYPE)
+    throw UDRException(38001,
+                       "Only support to generate series for exact numerics type");
```
Comment (DaveBirdsall) [inline core/sql/sqludr/SqlUdrPredefSeries.cpp]: Slightly better English might be "Only exact numeric types are supported"

File: core/sql/sqludr/SqlUdrPredefSeries.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  for (int i=0; i<paramCount; i++)
+    info.addFormalParameter(info.par().getColumn(i));
+
+  info.out().addLongColumn("ELEMENT");
+
+  // Set the function type, sessionize behaves like
+  // a reducer in MapReduce. Session ids are local
```
Comment (DaveBirdsall) [inline core/sql/sqludr/SqlUdrPredefSeries.cpp]: Comments should be updated.

File: core/sql/sqludr/SqlUdrPredefSeries.cpp [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    long end = info.par().getLong(1);
+    long step;
+    if (info.par().getNumColumns() == 2)
+      step = 1;
+    else
+      step = info.par().getLong(2);
+
+    if (0 == step)
+      throw UDRException(38001,
+                       "Step cannot be a zero");
```
Comment (DaveBirdsall) [inline core/sql/sqludr/SqlUdrPredefSeries.cpp]: Slightly better English would be "Step cannot be zero"

**Pull request merged**